src/console.cpp

```diff
@@ -178,11 +178,14 @@ bool addConsoleMessage(const char *Text, CONSOLE_TEXT_JUSTIFICATION jusType, SDW
 	time_t current_time;
 	time(&current_time);
 	struct tm time_info;
+#if defined(WZ_OS_WIN)
+	gmtime_s(&time_info, &current_time);
+#else
 	gmtime_r(&current_time, &time_info);
+#endif
 	char time_str[9];
 	ssprintf(time_str, "%d:%d", time_info.tm_hour, time_info.tm_min);
-	char buff2[2048]; // can be not enough to fit all message?
-	ssprintf(buff2, "[%s] %s", time_str, Text);
+	std::string buff2 = std::string("[") + time_str + "] " + Text;
```

Comment:
Is this line works correct?
`("[") + time_str + "] " + Text;`

Reply:
Yep!

See: [std::string::operator+=](https://www.cplusplus.com/reference/string/string/operator+=/)

```cpp
string& operator+= (const char* s);
```

The left-most is a `std::string` (`std::string("[")`), so this works through the operations in the line.